docs/design/test-results.md

```diff
@@ -0,0 +1,143 @@
+2. A view of recent test results updated every thirty minutes
+
+Option 1 is now available to everyone at
+http://storage.googleapis.com/kubernetes-test-history/static/index.html
+
+Option 2 is available to people inside of Google, but we will make it available
+to everyone in time for the release of 1.3
+
+![test result view](../images/test-results.png)
+
+## Detailed design
+
+We will take the tool we have access to internally and put it on a public
+website. Then we will update the tool to collect information from multiple
+sources.
+
+### Storing test result information
+
+* We will associate each jenkins jobs with a tab name
+* We will associate each jenkins job with a GCS prefix
+  * Example: gs://kubernetes-jenkins/logs/kubernetes-e2e-gce/
+  * The owner of this bucket will grant a specific service account read access to
+    the bucket as well as files under this prefix
+* Each run of the job will put files in a subfolder
+  * Example gs://kubernetes-jenkins/logs/kubernetes-e2e-gce/15039/
+  * Subfolders will have monotonically increasing numbers.
+  * This means that the next run id will be >= 15040
+* Each run will have the following files:
+  * Started.json
+      * Version string and timestamp (seconds since epoch) keys
+  * Finished.json
+      * Overall pass/fail result and timestamp keys
+  * artifacts/junit\_NN.xml where 01 <= NN <= 99
+      * Following standard junit.xml format outputted by jenkins
+
+### Collecitng test result information
+
+* Google will run a service that collects information about the list of jobs
+  * It will update this information every 30 minutes
+* It will then find the GCS prefix associated with that tab
+* It will then attempt to list objects under that prefix using the specified
+* service account
+* It will download the well known files under those prefixes:
+  * (started.json, finished.json and artifacts/junit\_NN.xml)
+  * Initially every object each time it runs
+* This information will allow it to maintain information about the state of test
+  results
+  * Jobs will have a list of runs sorted by id
+  * Each run will have
+    * An id
+    * A start time
+    * A duration
+    * A completion status (running, completed, failed)
+    * A list of test cases
+  * Each test case will have
+    * A name of the test case
+    * A start time
+    * A duration
+    * A completion status (passed, failed)
+
+
+#### Future optimization ideas
+
+* Only list new runs
+* Only download new and/or updated objects.
+* Download as fast as possible
+* Device a service for pushing results rather than polling.
+* Lots of opportunities to collect different/better information:
+  * example: failure stack trace
+
+### Displaying test result information
+
+* Google will run a service that displays the state of collected test results
+  * Available to all kubernetes developers
+* The service will display a list of tabs on the top of the screen
+  * Each tab is associated with a jenkins job
```

Comment:
may need to discuss which jobs are shown (there are a lot) and how non-Google jobs get in here